src/aggregator/client/conn.go

```diff
@@ -45,6 +50,27 @@ type sleepFn func(time.Duration)
 type connectWithLockFn func() error
 type writeWithLockFn func([]byte) error
 
+type connWriter interface {
+	io.Writer
+	Flush() error
+}
+
+type rawConnWriter struct {
+	writer io.Writer
+}
+
+func newRawConnWriter(w io.Writer) connWriter {
+	return &rawConnWriter{writer: w}
+}
+
+func (w *rawConnWriter) Write(p []byte) (int, error) {
+	return w.writer.Write(p)
+}
+
+func (w *rawConnWriter) Flush() error {
+	return nil
+}
```

Comment:
Maybe we should pull these into a common file, as well as `flateReader` and `flateConnReader`, then we can add a method that will build the correct readers and writers? Otherwise looks like there's a few places that are doing the same thing